### PR TITLE
Add support for distribution context manager and decorator

### PR DESCRIPTION
**Pull request by dnlserrano**
### What does this PR do?

Takes a stab at #563.

Introduces the ability to use a context manager or decorator for the custom [`DISTRIBUTION`](https://docs.datadoghq.com/developers/metrics/dogstatsd_metrics_submission/?tab=python#distribution) metric.

### Description of the Change

Introduces new `DistributedContextManagerDecorator` which allows one to use a new context manager or decorator (e.g., `statsd.distributed`). It does the same as `TimedContextManagerDecorator` (it is a subclass of it afterall), but changes the timing metric to use `statsd.distribution` instead of `statsd.timing`.

Thinking ahead, we do it by re-defining the `self.timing_func` used, which may later be passed in as an option to `statsd.timer`, as discussed [here](https://github.com/DataDog/datadogpy/issues/563).

### Alternate Designs

We could have gone with the global switch from the get go, but decided to be more conservative from the start following @hush-hush's [comment](https://github.com/DataDog/datadogpy/issues/563#issuecomment-650310530).

### Possible Drawbacks

Not sure if we can afford doing this change at the level of the general `context.py`. I'm not aware if this is used for Statsd-compatible metrics only or if it allows for changes regarding custom metrics, which is the case for `DISTRIBUTION`, as far as I'm aware. Appreciate any inputs there.

### Verification Process

Ran test with both `tox -e flake8` and `tox -e py36` locally. Also experimented with the context manager after installing the changes locally using `pip install -e <path_to_cloned_project_dir>`, following your [`DEVELOPMENT`](https://github.com/DataDog/datadogpy/blob/master/DEVELOPMENT.md#submit-your-changes) guide. Used [`jimf/statsd-logger`](https://github.com/jimf/statsd-logger) for quick-and-dirty experiment.

### Additional Notes

Python noob here, so please bear with. 🐻 

### Release Notes

Should be backwards compatible since it only introduces new behaviour, it does not break previous usage. Please double-check.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

**Comment by dnlserrano** [comment]
Is it possible to get a review here @hush-hush? 🙏

File: datadog/dogstatsd/context.py [hunk 1]
```diff
@@ -60,11 +62,25 @@ def _send(self, start):
...
+class DistributedContextManagerDecorator(TimedContextManagerDecorator):
+    """
+    A context manager and a decorator which will report the elapsed time in
+    the context OR in a function call using the custom distribution metric.
+    """
+    def __init__(self, statsd, metric=None, tags=None, sample_rate=1, use_ms=None):
+        if sys.version_info >= (3, 0):
+            super().__init__(statsd, metric, tags, sample_rate, use_ms)
+        else:
+            super(DistributedContextManagerDecorator, self).__init__(statsd, metric, tags, sample_rate, use_ms)
```
[the remark this answers is not in the document]
Comment (dnlserrano) [reply datadog/dogstatsd/context.py]: Done, thanks for the review @hush-hush.

**Comment by hush-hush** [comment]
Thanks @dnlserrano for the PR !

**Pull request merged**